### ISSUE TITLE
Bazel 4.0 breakage when using cgo

### ISSUE THREAD

**Issue by mrmeku** [bazelbuild/rules_go#2793]
### What version of rules_go are you using?

Replicated on 0.24.2 and 0.24.11

### What version of gazelle are you using?

Replicated on 0.21.1 and 0.22.3

### What version of Bazel are you using?

4.0.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Ubuntu x86

### What did you do?

Tried to build a go_library with `cgo` set to true and importing `cdeps`

### What did you expect to see?

A successful build

### What did you see instead?

```
ERROR: /usr/local/home/dan/.cache/bazel/_bazel_dan/5ba61dcdd7546c99c96c393c87400942/external/com_github_hillu_go_yara/BUILD.bazel:5:11: in go_library rule @com_github_hillu_go_yara//:go_default_library: 
Traceback (most recent call last):
        File "/usr/local/home/dan/.cache/bazel/_bazel_dan/5ba61dcdd7546c99c96c393c87400942/external/io_bazel_rules_go/go/private/rules/library.bzl", line 38, column 25, in _go_library_impl
                archive = go.archive(go, source)
        File "/usr/local/home/dan/.cache/bazel/_bazel_dan/5ba61dcdd7546c99c96c393c87400942/external/io_bazel_rules_go/go/private/actions/archive.bzl", line 82, column 28, in emit_archive
                cgo = cgo_configure(
        File "/usr/local/home/dan/.cache/bazel/_bazel_dan/5ba61dcdd7546c99c96c393c87400942/external/io_bazel_rules_go/go/private/rules/cgo.bzl", line 107, column 31, in cgo_configure
                cc_libs = _cc_libs(d)
        File "/usr/local/home/dan/.cache/bazel/_bazel_dan/5ba61dcdd7546c99c96c393c87400942/external/io_bazel_rules_go/go/private/rules/cgo.bzl", line 186, column 67, in _cc_libs
                libraries_to_link = as_iterable(target[CcInfo].linking_context.libraries_to_link)
Error: 'LinkingContext' value has no field or method 'libraries_to_link'
```

**Comment by mrmeku** [comment]
This appears to be a duplicate of https://github.com/bazelbuild/rules_go/issues/2720 which was closed by should be re-opened or used as a source of reference. For anyone encountering this bug you can set

`--incompatible_require_linker_input_cc_api=false` to fix rules_go bazel 4.0 compatibility

**Comment by jayconrod** [comment]
This issue was fixed in #2683.

Unfortunately #2683 cannot be backported to the 0.24 branch. The minimum supported version of Bazel on the 0.24 branch is 2.2.0, which did not support the new API.

This should be fixed on the 0.25 branch though, so you should be able to resolve the issue by upgrading.
